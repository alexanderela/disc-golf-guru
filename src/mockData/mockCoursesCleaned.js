const mockCoursesCleaned = [
  {
    id: '4712',
    name: 'Genesee Valley Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14620',
    address: '99 Elmwood Ave.',
    rating: '3.29',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '7628',
    name: 'Lookup Park',
    isFavorite: false,
    holes: '9',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14467',
    address: '3850 East Henrietta Rd.',
    rating: '3.13',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '4937',
    name: 'Monroe Community College DGC',
    isFavorite: false,
    holes: '18',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14623',
    address: '',
    rating: '2.94',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6795',
    name: 'Ellison Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14610',
    address: '',
    rating: '3.54',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6443',
    name: 'North Ponds DGC',
    isFavorite: false,
    holes: '9',
    city: 'Webster',
    state: 'NY',
    country: 'United States',
    zip: '14580',
    address: '750 Holt Rd.',
    rating: '1.75',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
];

export default mockCoursesCleaned;
