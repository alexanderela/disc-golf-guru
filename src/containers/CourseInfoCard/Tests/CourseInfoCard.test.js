import React from 'react';
import ReactDOM from 'react-dom';
import { CourseInfoCard } from '../';
import { shallow } from 'enzyme';
import mockCoursesCleaned from '../../../mockData/mockCoursesCleaned.js';

describe('CourseInfoCard', () => {
  let wrapper;
  let toggleFavorite;

  beforeEach(() => {
    toggleFavorite = jest.fn();
    const mockCourse = {
      address: "",
      city: "Brooklyn",
      country: "United States",
      holes: {header: "Number of holes: ", text: "18"},
      id: "1714",
      isFavorite: false,
      isPayToPlay: {header: "Pay to play: ", text: "No"},
      isPrivate: {header: "Private: ", text: "No"},
      name: "Prospect Park - Nethermead",
      rating: {header: "Rating: ", text: "3.25"},
      reviews: {header: "Reviews: ", text: "http://www.dgcoursereview.com/course.php?id=1714"},
      state: "NY",
      zip: "11215"
    };

    wrapper = shallow(
      <CourseInfoCard
        course={mockCourse}
        key={mockCourse.id}
        toggleFavorite={toggleFavorite}
        golfCourses={mockCoursesCleaned}
      />
    );
  });
  it('should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
