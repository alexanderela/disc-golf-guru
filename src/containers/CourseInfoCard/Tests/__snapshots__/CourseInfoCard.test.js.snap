// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseInfoCard should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseInfoCard
    course={
      Object {
        "address": "",
        "city": "Brooklyn",
        "country": "United States",
        "holes": Object {
          "header": "Number of holes: ",
          "text": "18",
        },
        "id": "1714",
        "isFavorite": false,
        "isPayToPlay": Object {
          "header": "Pay to play: ",
          "text": "No",
        },
        "isPrivate": Object {
          "header": "Private: ",
          "text": "No",
        },
        "name": "Prospect Park - Nethermead",
        "rating": Object {
          "header": "Rating: ",
          "text": "3.25",
        },
        "reviews": Object {
          "header": "Reviews: ",
          "text": "http://www.dgcoursereview.com/course.php?id=1714",
        },
        "state": "NY",
        "zip": "11215",
      }
    }
    golfCourses={
      Array [
        Object {
          "address": "99 Elmwood Ave.",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "4712",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Genesee Valley Park",
          "rating": "3.29",
          "state": "NY",
          "zip": "14620",
        },
        Object {
          "address": "3850 East Henrietta Rd.",
          "city": "Henrietta",
          "country": "United States",
          "holes": "9",
          "id": "7628",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Lookup Park",
          "rating": "3.13",
          "state": "NY",
          "zip": "14467",
        },
        Object {
          "address": "",
          "city": "Henrietta",
          "country": "United States",
          "holes": "18",
          "id": "4937",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Monroe Community College DGC",
          "rating": "2.94",
          "state": "NY",
          "zip": "14623",
        },
        Object {
          "address": "",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "6795",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Ellison Park",
          "rating": "3.54",
          "state": "NY",
          "zip": "14610",
        },
        Object {
          "address": "750 Holt Rd.",
          "city": "Webster",
          "country": "United States",
          "holes": "9",
          "id": "6443",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "North Ponds DGC",
          "rating": "1.75",
          "state": "NY",
          "zip": "14580",
        },
      ]
    }
    toggleFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-container"
        >
          <h2
            className="course-name"
          >
            Prospect Park - Nethermead
          </h2>
          <button
            className="favorite-btn"
            onClick={[Function]}
          >
            <i
              className="fas fa-heart"
            />
          </button>
        </div>,
        <div
          className="course-address"
        >
          <p
            className="address-fields"
          >
            
          </p>
          <p
            className="address-fields"
          >
            Brooklyn, NY 11215
          </p>
        </div>,
        <p
          className="course-info"
        >
          <span
            className="course-info-header"
          >
            Number of holes: 
          </span>
           
          18
        </p>,
        <p
          className="course-info"
        >
          <span
            className="course-info-header"
          >
            Rating: 
          </span>
           
          3.25
          /5
        </p>,
        <p
          className="course-info"
        >
          <span
            className="course-info-header"
          >
            Private: 
          </span>
           
          No
        </p>,
        <p
          className="course-info"
        >
          <span
            className="course-info-header"
          >
            Pay to play: 
          </span>
           
          No
        </p>,
        <p
          className="course-info"
        >
          <span
            className="course-info-header"
          >
            Reviews: 
          </span>
          <a
            href="http://www.dgcoursereview.com/course.php?id=1714"
            rel="noopener noreferrer"
            target="_blank"
          >
            Disc Golf Course Review
          </a>
        </p>,
      ],
      "className": "CourseInfoCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="course-name"
            >
              Prospect Park - Nethermead
            </h2>,
            <button
              className="favorite-btn"
              onClick={[Function]}
            >
              <i
                className="fas fa-heart"
              />
            </button>,
          ],
          "className": "header-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Prospect Park - Nethermead",
              "className": "course-name",
            },
            "ref": null,
            "rendered": "Prospect Park - Nethermead",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-heart"
              />,
              "className": "favorite-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-heart",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="address-fields"
            >
              
            </p>,
            <p
              className="address-fields"
            >
              Brooklyn, NY 11215
            </p>,
          ],
          "className": "course-address",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "address-fields",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brooklyn, NY 11215",
              "className": "address-fields",
            },
            "ref": null,
            "rendered": "Brooklyn, NY 11215",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="course-info-header"
            >
              Number of holes: 
            </span>,
            " ",
            "18",
          ],
          "className": "course-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Number of holes: ",
              "className": "course-info-header",
            },
            "ref": null,
            "rendered": "Number of holes: ",
            "type": "span",
          },
          " ",
          "18",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="course-info-header"
            >
              Rating: 
            </span>,
            " ",
            "3.25",
            "/5",
          ],
          "className": "course-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rating: ",
              "className": "course-info-header",
            },
            "ref": null,
            "rendered": "Rating: ",
            "type": "span",
          },
          " ",
          "3.25",
          "/5",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="course-info-header"
            >
              Private: 
            </span>,
            " ",
            "No",
          ],
          "className": "course-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Private: ",
              "className": "course-info-header",
            },
            "ref": null,
            "rendered": "Private: ",
            "type": "span",
          },
          " ",
          "No",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="course-info-header"
            >
              Pay to play: 
            </span>,
            " ",
            "No",
          ],
          "className": "course-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pay to play: ",
              "className": "course-info-header",
            },
            "ref": null,
            "rendered": "Pay to play: ",
            "type": "span",
          },
          " ",
          "No",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="course-info-header"
            >
              Reviews: 
            </span>,
            <a
              href="http://www.dgcoursereview.com/course.php?id=1714"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disc Golf Course Review
            </a>,
          ],
          "className": "course-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reviews: ",
              "className": "course-info-header",
            },
            "ref": null,
            "rendered": "Reviews: ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Disc Golf Course Review",
              "href": "http://www.dgcoursereview.com/course.php?id=1714",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Disc Golf Course Review",
            "type": "a",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-container"
          >
            <h2
              className="course-name"
            >
              Prospect Park - Nethermead
            </h2>
            <button
              className="favorite-btn"
              onClick={[Function]}
            >
              <i
                className="fas fa-heart"
              />
            </button>
          </div>,
          <div
            className="course-address"
          >
            <p
              className="address-fields"
            >
              
            </p>
            <p
              className="address-fields"
            >
              Brooklyn, NY 11215
            </p>
          </div>,
          <p
            className="course-info"
          >
            <span
              className="course-info-header"
            >
              Number of holes: 
            </span>
             
            18
          </p>,
          <p
            className="course-info"
          >
            <span
              className="course-info-header"
            >
              Rating: 
            </span>
             
            3.25
            /5
          </p>,
          <p
            className="course-info"
          >
            <span
              className="course-info-header"
            >
              Private: 
            </span>
             
            No
          </p>,
          <p
            className="course-info"
          >
            <span
              className="course-info-header"
            >
              Pay to play: 
            </span>
             
            No
          </p>,
          <p
            className="course-info"
          >
            <span
              className="course-info-header"
            >
              Reviews: 
            </span>
            <a
              href="http://www.dgcoursereview.com/course.php?id=1714"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disc Golf Course Review
            </a>
          </p>,
        ],
        "className": "CourseInfoCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="course-name"
              >
                Prospect Park - Nethermead
              </h2>,
              <button
                className="favorite-btn"
                onClick={[Function]}
              >
                <i
                  className="fas fa-heart"
                />
              </button>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Prospect Park - Nethermead",
                "className": "course-name",
              },
              "ref": null,
              "rendered": "Prospect Park - Nethermead",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-heart"
                />,
                "className": "favorite-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-heart",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="address-fields"
              >
                
              </p>,
              <p
                className="address-fields"
              >
                Brooklyn, NY 11215
              </p>,
            ],
            "className": "course-address",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "address-fields",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brooklyn, NY 11215",
                "className": "address-fields",
              },
              "ref": null,
              "rendered": "Brooklyn, NY 11215",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="course-info-header"
              >
                Number of holes: 
              </span>,
              " ",
              "18",
            ],
            "className": "course-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Number of holes: ",
                "className": "course-info-header",
              },
              "ref": null,
              "rendered": "Number of holes: ",
              "type": "span",
            },
            " ",
            "18",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="course-info-header"
              >
                Rating: 
              </span>,
              " ",
              "3.25",
              "/5",
            ],
            "className": "course-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rating: ",
                "className": "course-info-header",
              },
              "ref": null,
              "rendered": "Rating: ",
              "type": "span",
            },
            " ",
            "3.25",
            "/5",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="course-info-header"
              >
                Private: 
              </span>,
              " ",
              "No",
            ],
            "className": "course-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Private: ",
                "className": "course-info-header",
              },
              "ref": null,
              "rendered": "Private: ",
              "type": "span",
            },
            " ",
            "No",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="course-info-header"
              >
                Pay to play: 
              </span>,
              " ",
              "No",
            ],
            "className": "course-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pay to play: ",
                "className": "course-info-header",
              },
              "ref": null,
              "rendered": "Pay to play: ",
              "type": "span",
            },
            " ",
            "No",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="course-info-header"
              >
                Reviews: 
              </span>,
              <a
                href="http://www.dgcoursereview.com/course.php?id=1714"
                rel="noopener noreferrer"
                target="_blank"
              >
                Disc Golf Course Review
              </a>,
            ],
            "className": "course-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reviews: ",
                "className": "course-info-header",
              },
              "ref": null,
              "rendered": "Reviews: ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Disc Golf Course Review",
                "href": "http://www.dgcoursereview.com/course.php?id=1714",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Disc Golf Course Review",
              "type": "a",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
